meson.build

```diff
@@ -0,0 +1,39 @@
+project(
+  'slingshot',
+  'vala', 'c',
+  version : '2.1.2'
+)
+
+i18n = import('i18n')
+
+add_project_arguments(
+    '-DGETTEXT_PACKAGE="@0@"'.format(meson.project_name()),
+    language: 'c'
+)
+
+add_project_arguments(
+    ['--vapidir', join_paths(meson.current_source_dir(), 'vapi')],
+    language: 'vala'
+)
+
```

Comment:
by convention we all the dependencies here so we can control them in one place only
(things like `glib_dep = dependency('glib-2.0')`)